pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -494,6 +533,45 @@ func (c *VSphereController) updateConditions(
 	return nil
 }
 
+func (c *VSphereController) addRequiresAdminAck(ctx context.Context, lastCheckResult checks.ClusterCheckResult, name string) error {
+	adminGate, err := c.managedConfigMapLister.ConfigMaps(managedConfigNamespace).Get(adminGateConfigMap)
+	if err != nil {
+		return fmt.Errorf("failed to get admin-gate configmap: %v", err)
+	}
+
+	klog.V(2).Infof("Updating admin-gates to require admin-ack for vSphere CSI migration")
+
+	_, ok := adminGate.Data[migrationAck413]
+	if !ok {
+		adminGate.Data[migrationAck413] = "vSphere CSI migration will be enabled in Openshift-4.14. Your cluster appears to be using in-tree vSphere volumes and is on a vSphere version that has CSI migration related bugs. See - https://access.redhat.com/node/7011683 for more information, before upgrading to 4.14."
+
+	}
+	_, _, err = resourceapply.ApplyConfigMap(ctx, c.kubeClient.CoreV1(), c.eventRecorder, adminGate)
+	return err
+}
+
+func (c *VSphereController) removeAdminAck(ctx context.Context, name string) error {
+	adminGate, err := c.managedConfigMapLister.ConfigMaps(managedConfigNamespace).Get(adminGateConfigMap)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get admin-gate configmap: %v", err)
+	}
+
+	klog.V(2).Infof("removing admin-gates that is required for CSI migration")
```

Comment:
I would log it below, when the key exists and is actually removed. Esp. at level 2.

Reply:
moved.